cmd/kola/testiso.go

```diff
@@ -0,0 +1,327 @@
+	if err := doTestIso(cmd, args); err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", err)
+		os.Exit(1)
+	}
+}
+
+func doTestIso(cmd *cobra.Command, args []string) error {
+	//var err error
+
+	if kola.CosaBuild == nil {
+		return fmt.Errorf("Must provide --cosa-build")
+	}
+
+	if kola.CosaBuild.BuildArtifacts.Kernel.Path != "" {
+		return errors.Wrapf(testLegacyInstaller(), "testing legacy installer")
+	} else {
+		return fmt.Errorf("build %s has no installer kernel", kola.CosaBuild.Name)
+	}
+}
+
+func testLegacyInstaller() error {
+	builddir := filepath.Dir(kola.Options.CosaBuild)
+	metalimg := kola.CosaBuild.BuildArtifacts.Metal.Path
+
+	builder := platform.NewBuilder(kola.QEMUOptions.Board, "")
+	defer builder.Close()
+	builder.Firmware = kola.QEMUOptions.Firmware
+	builder.AddDisk(&platform.Disk{
+		Size: "12G", // Arbitrary
+	})
+	// Arbitrary
+	builder.Memory = 1536
+
+	builder.InheritConsole = true
+
+	tftpipaddr := "192.168.76.2"
+	var boottype, networkdevice, bootindex, pxeimagepath string
+	switch kola.QEMUOptions.Board {
+	case "amd64-usr":
+		boottype = "pxe"
+		networkdevice = "e1000"
+		pxeimagepath = "/usr/share/syslinux/"
+		break
+	case "ppc64le-usr":
+		boottype = "grub"
+		networkdevice = "virtio-net-pci"
+		break
+	case "s390x-usr":
+		boottype = "pxe"
+		networkdevice = "virtio-net-ccw"
+		tftpipaddr = "10.0.2.2"
+		bootindex = "1"
+	default:
+		return fmt.Errorf("Unsupported arch %s", kola.QEMUOptions.Board)
+	}
+
+	unit := `[Unit]
+	Requires=dev-virtio\\x2dports-completion.device
+	OnFailure=emergency.target
+	OnFailureJobMode=isolate
+	[Service]
+	Type=oneshot
+	ExecStart=/bin/sh -c '/usr/bin/echo coreos-installer-test-OK >/dev/virtio-ports/completion && systemctl poweroff'
+	[Install]
+	RequiredBy=multi-user.target
+`
+	config := ignv3types.Config{
+		Ignition: ignv3types.Ignition{
+			Version: "3.0.0",
+		},
+		Systemd: ignv3types.Systemd{
+			Units: []ignv3types.Unit{
+				{
+					Name:     "coreos-test-installer.service",
+					Contents: &unit,
+					Enabled:  util.BoolToPtr(true),
+				},
+			},
+		},
+	}
+	var serializedConfig []byte
+	if sdk.TargetIgnitionVersionFromName(metalimg) == "v2" {
+		ignc2, err := ignconverter.Translate3to2(config)
+		if err != nil {
+			return err
+		}
+		buf, err := json.Marshal(ignc2)
+		if err != nil {
+			return err
+		}
+		serializedConfig = buf
+	} else {
+		buf, err := json.Marshal(config)
+		if err != nil {
+			return err
+		}
+		serializedConfig = buf
+	}
+
+	tempdir, err := ioutil.TempDir("", "kola-testiso")
+	if err != nil {
+		return err
+	}
+
+	tftpdir := filepath.Join(tempdir, "tftp")
+	if err := os.Mkdir(tftpdir, 0777); err != nil {
+		return err
+	}
+	defer os.RemoveAll(tftpdir)
+
+	if err := ioutil.WriteFile(filepath.Join(tftpdir, "config.ign"), serializedConfig, 0777); err != nil {
+		return err
+	}
+
+	metalIsCompressed := !strings.HasSuffix(metalimg, ".raw")
+	metalname := metalimg
+	if !metalIsCompressed {
+		fmt.Println("Compressing metal image")
```

Comment:
For the `Println` log statements might be worthwhile importing `plog` & using that so it can be easily suppressed.

Reply:
I looked at this and spent 5 minutes banging my head against understanding the log levels and details like whether `\n` would be automatically emitted or not but...the bigger problem here is that we output the qemu console by default, and we kind of need to at the moment because we don't have the journal handling hooked up.

So...that can come next?